### PR TITLE
Basic index route

### LINKED ISSUE #1526: SPIKE: Investigate adding an index route to render home pages
**Is your feature request related to a problem? Please describe.**
We should investigate adding an index route so we can render service home pages e.g. `/news` & `/mundo`

**Describe the solution you'd like**
We should look into generalising `getInitialData` and construct the data URL based on the page type to fetch data from a single `/data` endpoint.

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

**Additional context**

- [ ] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by chris-hinds**
Resolves #1526 

**Overall change:** _This PR enables a separate Index route with an index container, including fixture data for the news service front_page._

**Code changes:**

- Adds an index (home page) route
- Moves `getInitialData` to `getArticleData`
- Implements `getIndexData`
- Adds a basic Index container to simply output the top story titles

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/server/index.jsx [hunk 1]
```diff
@@ -123,6 +149,21 @@ server
       }
     });
   })
+  .get(homePageRegexPath, async ({ url, headers }, res) => {
```
Comment (dr3) [inline src/server/index.jsx:152]: Just being nosey, but we should aim to use `/*` as our route here and let react router handle it all :)

Comment (chris-hinds) [reply src/server/index.jsx:152]: Agree, although that is not in place at the moment right?

Comment (dr3) [reply src/server/index.jsx:152]: Nope, we had to make it used a regex as before when we used @jtart's React universal app, the error thrown for missing a route https://github.com/bbc/simorgh/blob/latest/src/app/routes/loadInitialData/index.js#L11 couldn't be distinguished between a route error and a actual error

We should handle it differently so they can be distinguished

File: src/app/containers/HomePage/index.jsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import GlobalStyle from '../../lib/globalStyles';
+import nodeLogger from '../../helpers/logger.node';
+import ConsentBanner from '../ConsentBanner';
+
+const logger = nodeLogger(__filename);
+
+/*
+  [1] This handles async data fetching, and a 'loading state', which we should look to handle more intelligently.
+*/
+const HomePageContainer = ({ loading, error, data, bbcOrigin }) => {
```
Comment (dr3) [inline src/app/containers/HomePage/index.jsx:17]: Seeing as these are likely to have very similar `loading`, `error` and `data` handling, could we maybe share some logic?

Comment (chris-hinds) [reply src/app/containers/HomePage/index.jsx:17]: Yup, we should have a dedicated Loading component/container and also too what else we can combine.

File: src/app/routes/getIndexData/index.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    typeof window !== 'undefined' &&
+    window.location &&
+    window.location.origin
+  ) {
+    baseUrl = getBaseUrl(window.location.origin);
+  }
+
+  const url = `${baseUrl}/${service}/index/front_page.json`;
+
+  try {
```
Comment (dr3) [inline src/app/routes/getIndexData/index.js:25]: I know its POC but we should share allllll the thingssss

Comment (chris-hinds) [reply src/app/routes/getIndexData/index.js:25]: This is why originally I was passing a pageType props to `getInitialData` and was constructing the url based off of that pageType value e.e. `article` or `index`.

However you mentioned the preProcessor  and I am not sure how that overlaps with `getInitial` data yet.

With this setup I personally do not see the point in have `getArticleData` and `getIndexData` when 98% of the code is the same

File: src/app/routes/getIndexData/index.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  if (
+    typeof window !== 'undefined' &&
+    window.location &&
+    window.location.origin
+  ) {
+    baseUrl = getBaseUrl(window.location.origin);
+  }
+
+  const url = `${baseUrl}/${service}/index/front_page.json`;
```
Comment (chris-hinds) [inline src/app/routes/getIndexData/index.js:23]: An alternative to duplicating 98% of the original `getInitialData` method is to keep that and introduce a page type prop allowing us to construct the url based on the page type.

**Comment by jamesbhobbs** [comment]
@hindsc52 can you ensure there's an issue for this and link it please. Even spikes should have issues.

**Comment by chris-hinds** [comment]
> @hindsc52 can you ensure there's an issue for this and link it please. Even spikes should have issues.

@benjaminhobbs https://github.com/bbc/simorgh/issues/1526

**Comment by jamesbhobbs** [comment]
PR closed and reopened to trigger Travis PR build

**Pull request closed without merging**